x/downtime-detector/abci.go

```diff
@@ -0,0 +1,33 @@
+package downtimedetector
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/osmosis-labs/osmosis/v13/x/downtime-detector/types"
+)
+
+func (k *Keeper) BeginBlock(ctx sdk.Context) {
+	curTime := ctx.BlockTime()
+	lastBlockTime, err := k.GetLastBlockTime(ctx)
+	if err != nil {
+		ctx.Logger().Error("Downtime-detector, could not get last block time, did initialization happen correctly. " + err.Error())
+	}
+	downtime := curTime.Sub(lastBlockTime)
+	if downtime > 30*time.Second {
```

Comment:
nit: shouldn't we just do `k.saveDowntimeUpdates(ctx, downtime)` here and have that function determine when to save the downtime or not ?

Reply:
Good point, will change to do that